ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Trace.hs

```diff
@@ -471,6 +453,33 @@ terseGDDEvent = \case
         showIdling | idling = ", idling"
                    | otherwise = ""
 
+    showPeers :: Map.Map PeerId String -> [String]
+    showPeers = fmap (\ (peer, v) -> "        " ++ condense peer ++ ": " ++ v) . Map.toList
+
+-- * Other utilities
+terseGDDEvent :: TraceGDDEvent PeerId TestBlock -> String
+terseGDDEvent = \case
+  TraceGDDEvent {sgen = GenesisWindow sgen, curChain, bounds, candidates, candidateSuffixes, losingPeers, loeHead} ->
+    unlines $ [
+      "GDG | Window: " ++ window sgen loeHead,
+      "      Selection: " ++ terseHFragment curChain,
+      "      Candidates:"
+      ] ++
+      showPeers (either (const "G") terseHeader . AF.head <$> candidates) ++
+      [
+      "      Candidate suffixes (bounds):"
+      ] ++
+      showPeers (terseHFragment . clippedFragment <$> bounds) ++
+      ["      Density bounds:"] ++
+      prettyDensityBounds bounds ++
+      ["      New candidate tips:"] ++
+      showPeers (tersePoint . castPoint <$> Map.map AF.headPoint candidateSuffixes) ++
+      [
+        "      Losing peers: " ++ show losingPeers,
+      "      Setting loeFrag: " ++ terseAnchor (AF.castAnchor loeHead)
+      ]
+  where
+
```

Comment:
I'm not getting what is the difference in this diff, and I'm seeing that some conflict is going to appear here with csj related changes.

Reply:
I think I get it after some more staring.

Reply:
just that `prettyDensityBounds` was pulled out